packages/fiori/src/SideNavigationItemBase.ts

```diff
@@ -0,0 +1,241 @@
+	 * @public
+	 * @type {string}
+	 * @defaultvalue ""
+	 * @name sap.ui.webc.fiori.SideNavigationItemBase.prototype.icon
+	 */
+	@property()
+	icon!: string;
+
+	/**
+	 * Defines whether the item is selected
+	 *
+	 * @public
+	 * @type {boolean}
+	 * @defaultvalue false
+	 * @name sap.ui.webc.fiori.SideNavigationItemBase.prototype.selected
+	 */
+	@property({ type: Boolean })
+	selected!: boolean;
+
+	/**
+	 * Defines the link target URI. Supports standard hyperlink behavior.
+	 * If a JavaScript action should be triggered,
+	 * this should not be set, but instead an event handler
+	 * for the <code>click</code> event should be registered.
+	 *
+	 * @public
+	 * @type {string}
+	 * @defaultvalue ""
+	 * @name sap.ui.webc.fiori.SideNavigationItemBase.prototype.href
+	 * @since 1.19.0
+	 */
+	@property()
+	href!: string;
+
+	/**
+	 * Defines the component target.
+	 * <br><br>
+	 * <b>Notes:</b>
+	 *
+	 * <ul>
+	 * <li><code>_self</code></li>
+	 * <li><code>_top</code></li>
+	 * <li><code>_blank</code></li>
+	 * <li><code>_parent</code></li>
+	 * <li><code>_search</code></li>
+	 * </ul>
+	 *
+	 * <b>This property must only be used when the <code>href</code> property is set.</b>
+	 *
+	 * @public
+	 * @type {string}
+	 * @defaultvalue ""
+	 * @name sap.ui.webc.fiori.SideNavigationItemBase.prototype.target
+	 * @since 1.19.0
+	 */
+	@property()
+	target!: string;
+
+	/**
+	 * Defines whether the component is disabled.
+	 * A disabled component can't be pressed or
+	 * focused, and it is not in the tab chain.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.SideNavigationItemBase.prototype.disabled
+	 * @defaultvalue false
+	 * @public
+	 * @since 1.19.0
+	 */
+	@property({ type: Boolean })
+	disabled!: boolean;
+
+	/**
+	 * Defines the tooltip of the component.
+	 * @type {string}
+	 * @defaultvalue ""
+	 * @private
+	 * @name sap.ui.webc.fiori.SideNavigationItemBase.prototype.title
+	 * @since 1.0.0-rc.16
+	 */
+	@property()
+	title!: string;
+
+	@property({ defaultValue: "-1", noAttribute: true })
+	_tabIndex!: string;
+
+	get _tooltip() {
+		return this.title || this.text;
+	}
+
+	get _href() {
+		return (!this.disabled && this.href) ? this.href : undefined;
+	}
+
+	get _target() {
+		return (!this.target && this.target) ? this.target : undefined;
+	}
+
+	get _selected() {
+		return this.selected;
+	}
+
+	get classesArray() {
+		const classes = [];
+
+		if (this.disabled) {
+			classes.push("ui5-sn-item-disabled");
+		}
+
+		if (this._selected) {
+			classes.push("ui5-sn-item-selected");
+		}
+
+		return classes;
+	}
+
+	get _classes() {
+		return this.classesArray.join(" ");
+	}
+
+	get _ariaCurrent() {
+		if (!this.selected) {
+			return undefined;
+		}
+
+		return "page";
+	}
+
+	get _effectiveTabIndex() {
+		if (this.disabled) {
+			return undefined;
+		}
+
+		return this._tabIndex;
+	}
+
+	get sideNavigation() {
+		let parentElement = this.parentElement;
+
+		while (parentElement) {
+			if (parentElement instanceof SideNavigation) {
```

Comment:
Avoid a circular dependency here by checking if the parentElement.hasAttribute("ui5-side-navigation"). Then you could remove the SideNavigation import.

Reply:
Now SideNavigation is imported as a type.

Reply:
This can't work during execution of the code. After the TS compliation step completes, all "import type" declarations get removed from the code. Check the build output inside the `dist/` folder to verify it.